src/app/components/sites/site/site.component.ts

```diff
@@ -22,17 +26,84 @@ class SiteComponent extends PageComponent implements OnInit {
   @Input() public site: Site;
   public defaultDescription = "<i>No description found</i>";
   public recordings: AudioRecording[];
-  public recordingsEnd: DateTimeTimezone;
-  public recordingsStart: DateTimeTimezone;
+  public recordingsEnd: DateTime;
+  public recordingsStart: DateTime;
   public marker: List<MapMarkerOption>;
+  public annotations: AudioEvent[];
 
-  constructor() {
+  constructor(
+    private audioEventsApi: ShallowAudioEventsService,
+    private audioRecordingsApi: AudioRecordingsService
+  ) {
     super();
   }
 
   public ngOnInit() {
-    this.recordings = [];
     this.marker = sanitizeMapMarkers(this.site.getMapMarker());
+
+    this.getAnnotations();
+    this.getNewestDates();
+    this.getOldestDates();
+  }
+
+  public humanizeDate(date: DateTime) {
+    return date ? date.toLocaleString(DateTime.DATETIME_FULL) : "(loading)";
+  }
+
+  private getAnnotations() {
+    this.audioEventsApi
+      .filter({
+        filter: { ["audio_recordings.site_id"]: { eq: this.site.id } } as any,
+        sorting: { orderBy: "updatedAt", direction: "desc" },
+      })
+      .pipe(takeUntil(this.unsubscribe))
+      .subscribe(
+        (events) => {
+          // Limit the selection of audio events by tagging count
+          let count = 0;
+          const maxTags = 10;
+          this.annotations = [];
+          for (const event of events) {
+            this.annotations.push(event);
+            count++;
+
+            if (count > maxTags) {
+              return;
+            }
+          }
+        },
```

Comment:
this should be a pretty sraight forward functional map? why the foor loop?

`takeUntil` or something like that

Reply:
Whoops, while this is needed i didnt implement it properly. It should be attempting to limit the number of events pushed to the annotations array so that it stops adding tags around 10.